info.php

```diff
@@ -121,231 +52,318 @@ function check_if_wiris_button_are_in_atto_toolbar() {
 
-// Filter version existance test.
-echo create_filter_version_row($solutionlink);
+    // Condition 3 - filter version same as one of enabled editor
+    $sameversion = false;
 
-// Filter enabled.
-echo create_filter_enabled_row($solutionlink);
+    foreach ($enabledpluginsversion as $pluginversion) {
+        if ($pluginversion == $filterversion) {
+            $sameversion = true;
+        }
+    }
 
-// Atto is installed.
-if (create_atto_installed_row($currenteditordata, $solutionlink)) {
-    // Version compatibility test.
-    echo create_atto_compatibility_row($currenteditordata, $solutionlink);
+    $instalationsuccess = ($editorinstalledenabled && $filterenabled && $sameversion);
 
-    // MathType enabled test.
-    echo create_atto_enabled_row($currenteditordata, $solutionlink);
+    return $instalationsuccess;
 }
 
-// Tiny is installed.
-echo create_tiny_installed_row($solutionlink);
+// Page prologue.
+$PAGE->set_context(context_system::instance());
+$PAGE->set_title(get_string('title', 'filter_wiris'));
+$PAGE->set_url('/filter/wiris/info.php', array());
 
-// Close table.
-echo create_table_close();
+echo $OUTPUT->header();
+warning_tiny_incompatibility();
+
+global $CFG;
+$plugin = new stdClass();
+require($CFG->dirroot . '/filter/wiris/version.php');
+// Array of arrays witch contains all necesaty information for testing
+$plugins = filter_wiris_pluginwrapper::get_wiris_plugins_information();
+
+// Save version of all enabled plugins for test
+$enabledplugins = [];
+
+// Start to check all tests
+$moodleversion = get_moodle_version();
+$existsfilter = get_exists_mt_filter($plugins);
+$filterversion = get_mt_filter_version($existsfilter, $plugins);
+$filterenabled = get_mt_filter_enabled($existsfilter);
+
+$exisatto = get_exists_editor('atto');
+$existsmtatto = get_exists_mt_editor($exisatto, $plugins, 'atto');
+$mtattoversion = get_mt_editor_version($existsmtatto, $plugins, 'atto');
+$mtattoenabled = get_mt_editor_enabled($existsmtatto, 'atto');
+$enabledplugins['atto'] = ($mtattoenabled) ? $mtattoversion : null;
+
+$existstinylegacy = get_exists_editor('tinymce');
+$existsmttinylegacy = get_exists_mt_editor($existstinylegacy,  $plugins, 'tinymce');
+$mttinylegacyversion = get_mt_editor_version($existsmttinylegacy,  $plugins, 'tinymce');
+$mttinylegacyenabled = get_mt_editor_enabled($existsmttinylegacy, 'tinymce');
+$enabledplugins['tinymce'] = ($mttinylegacyenabled) ? $mttinylegacyversion : null;
+
+$existstinycurrent = get_exists_editor('tiny');
+$existsmttinycurrent = get_exists_mt_editor($existstinycurrent, $plugins, 'tiny');
+$mttinycurrentversion = get_mt_editor_version($existsmttinycurrent, $plugins, 'tiny');
+$mttinycurrentenabled = get_mt_editor_enabled($existsmttinycurrent, 'tiny');
+$enabledplugins['tiny'] = ($mttinycurrentenabled) ? $mttinycurrentversion : null;
+
+$instalationresult = get_instalation_check($filterenabled,$filterversion, $enabledplugins);
+
+// Construct table
+table_open();
```

Comment:
I'm not sure if maybe the name needs to be more clear about the action (construct table).

Reply:
Maybe the construct is a bit strange...how do you think about renaming:

1. `table_open` to `start_table`.
2. `create_table_row` to `add_table_row`.
3. `table_close` to `end_table`.